### PR TITLE
Add ruby 3 support

### PR DESCRIPTION
**Pull request by kwong-yw**
* Add Ruby 3 support (notably changes due to keyword arg support etc)

We're reviving this fork as the original repo, while still receiving new updates, has not published a new gem since. This'll allow us to publish our own release as-needed.

### CODE REVIEW

File: sneakers.gemspec [hunk 1]
```diff
@@ -33,7 +33,7 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency 'rabbitmq_http_api_client'
   gem.add_development_dependency 'redis'
 
-  gem.add_development_dependency 'minitest', '~> 5.15'
+  gem.add_development_dependency 'minitest', '~> 5.15.0'
```
Comment (kwong-yw) [inline sneakers.gemspec:36]: It seems like newer versions of `minitest` was causing specs to fail in Ruby 2.6 and 2.7. For now, if we just restrict the `minitest` version specs will continue to pass, and Ruby 3.0 compatibility is not broken.

**Comment by ancharon** [comment]
This just brings this fork in line with the mainline, correct? I.e. this doesn't yet address the rake incompatibility?

**Comment by kwong-yw** [comment]
> This just brings this fork in line with the mainline, correct? I.e. this doesn't yet address the rack incompatibility?

I happened to click a button that just brought this fork up to date with mainline here: https://github.com/Yesware/sneakers/commit/32c0d50abb6b3b72eaec98d7b0995946f09834bd

So the Rack support has already been fixed 👀 

See: 
image

**Comment by ancharon** [comment]
Whoops, rake not rack. Updated my comment above.

> > This just brings this fork in line with the mainline, correct? I.e. this doesn't yet address the rack incompatibility?
> 
> I happened to click a button that just brought this fork up to date with mainline here: [32c0d50](https://github.com/Yesware/sneakers/commit/32c0d50abb6b3b72eaec98d7b0995946f09834bd)
> 
> So the Rack support has already been fixed 👀
> 
> See: image

🤔 Hmm, okay. Looks like there was one commit that relaxed the rake dependency for production builds, and then the dev dependency forcing rake v12 was removed in a separate commit, so that should be all set. 👍 I didn't see that second commit resolving the dev side earlier.

**Review by ancharon — APPROVED** [review]
g2g

**Review by sebastianfairchild — APPROVED** [review]

**Pull request merged**